gtsfm/bundle/bundle_adjustment.py

```diff
@@ -191,34 +210,30 @@ def run(
         # construct the results
         optimized_data = values_to_gtsfm_data(result_values, initial_data, self._shared_calib)
 
-        def get_metrics_from_sfm_data(sfm_data: GtsfmData, suffix: str) -> List[GtsfmMetric]:
-            """Helper to get bundle adjustment metrics from a GtsfmData object with a suffix for metric names."""
-            metrics = []
-            metrics.append(GtsfmMetric(name="number_cameras", data=len(sfm_data.get_valid_camera_indices())))
-            metrics.append(GtsfmMetric("number_tracks" + suffix, sfm_data.number_tracks()))
-            metrics.append(
-                GtsfmMetric(
-                    "3d_track_lengths" + suffix,
-                    sfm_data.get_track_lengths(),
-                    plot_type=GtsfmMetric.PlotType.HISTOGRAM,
-                )
-            )
-            metrics.append(GtsfmMetric(f"reprojection_errors{suffix}_px", sfm_data.get_scene_reprojection_errors()))
-            return metrics
-
-        ba_metrics = GtsfmMetricsGroup(
-            name="bundle_adjustment_metrics", metrics=get_metrics_from_sfm_data(optimized_data, suffix="_unfiltered")
-        )
         logger.info("[Result] Number of tracks before filtering: %d", optimized_data.number_tracks())
 
         # filter the largest errors
-        filtered_result = optimized_data.filter_landmarks(self._output_reproj_error_thresh)
+        if self._output_reproj_error_thresh:
+            filtered_result = optimized_data.filter_landmarks(self._output_reproj_error_thresh)
+        else:
+            filtered_result = optimized_data
+
+        logger.info("[Result] Number of tracks after filtering: %d", filtered_result.number_tracks())
+
+        return optimized_data, filtered_result
+
+    def evaluate(
+        self, unfiltered_data: GtsfmData, filtered_data: GtsfmData, cameras_gt: List[PinholeCameraCal3Bundler] = None
+    ) -> GtsfmMetricsGroup:
+        ba_metrics = GtsfmMetricsGroup(
+            name=METRICS_GROUP, metrics=metrics_utils.get_stats_for_sfmdata(unfiltered_data, suffix="_unfiltered")
+        )
 
         if cameras_gt is not None:
             poses_gt = [cam.pose() for cam in cameras_gt]
 
             # align the sparse multi-view estimate after BA to the ground truth pose graph.
-            filtered_result = filtered_result.align_via_Sim3_to_poses(wTi_list_ref=poses_gt)
+            filtered_result = filtered_data.align_via_Sim3_to_poses(wTi_list_ref=poses_gt)
```

Comment:
please consider renaming filtered_result to filtered_data_aligned and use the same in line 251.